.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and test
+
+on:
+  merge_group:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        buildtype: [debug, release]
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Update apt repositories for ccache
+        run: sudo apt update
+      - name: Set up ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          key: ccache-linux-${{ matrix.buildtype }}
+      - name: Setup compiler and build tools
+        run: sudo apt install --no-install-recommends --yes g++-12 meson ninja-build
+      - name: Configure
+        run: |
+            CXX="ccache g++-12" meson setup \
+              --buildtype ${{ matrix.buildtype }} \
+              --warnlevel 3 \
+              --werror \
+              -Dcatch2:werror=false \
+              -Dcli11:werror=false \
+              -Dfmt:werror=false \
+              -Dnlohmann_json:werror=false \
+              -Dsqlite3:werror=false \
```

Comment:
I don't know enough about meson to say if this is the nicest way of doing this. It'd be nice if dependencies simply had `-Wno-error` by default. It seems like it's also possible to set the default to `false` in `meson.build` with
```diff
diff --git a/meson.build b/meson.build
index 1224c7d..92231fb 100644
--- a/meson.build
+++ b/meson.build
@@ -15,11 +15,11 @@ conf_data = configuration_data()
 #
 # use meson wrap to provide the dependencies, because the 'default_library=static' option
 # will be propogated to each dependency, for a statically linked executable.
-catch_dep = subproject('catch2').get_variable('catch2_with_main_dep')
-cli11_dep = subproject('cli11').get_variable('CLI11_dep')
-fmt_dep   = subproject('fmt').get_variable('fmt_dep')
-json_dep = subproject('nlohmann_json').get_variable('nlohmann_json_dep')
-sqlite3_dep = subproject('sqlite3').get_variable('sqlite3_dep')
+catch_dep = subproject('catch2', default_options: 'werror=false').get_variable('catch2_with_main_dep')
+cli11_dep = subproject('cli11', default_options: 'werror=false').get_variable('CLI11_dep')
+fmt_dep   = subproject('fmt', default_options: 'werror=false').get_variable('fmt_dep')
+json_dep = subproject('nlohmann_json', default_options: 'werror=false').get_variable('nlohmann_json_dep')
+sqlite3_dep = subproject('sqlite3', default_options: 'werror=false').get_variable('sqlite3_dep')

 # the lib dependency is all of the common funtionality shared between the CLI
 # and the slurm plugin.
```
Do you have a preference or better suggestion here?

Reply:
I don't have a strong preference.

Adding the `werror=false` to the `meson.build` file is a bit less fragile, maybe? Because every time a new dependency is added, we would need to update this GitHub action yaml file to pass `werror=false` to that dependency.

Reply:
Yeah, I'd agree with that assessment. Adding the flags in the CI configuration was just the first thing I tried. Changed in https://github.com/bcumming/uenv2/pull/2/commits/91f6ee63362f199f79967299107edcc3bfa58aad.